### ISSUE TITLE
CodeStream 12.7 causes JetBrains IDEs on Linux to freeze

### ISSUE THREAD

**Issue by quicksketch** [TeamCodeStream/codestream#836]
Tried out CodeStream for the first time today and didn't have the best of luck. I installed the CodeStream plugin through PHPStorm (File -> Settings -> Plugins), restarted. PHPStorm tries to open my last open project (a Drupal 9 installation) but it fails at "Loading components"

![image](https://user-images.githubusercontent.com/100206/151291509-284b3f8e-0b24-4ed4-b31b-23bf8d8e3aa2.png)

At this point the IDE is completely locked up and I can only kill it with `killall java`.

Here's the dump from running it from the command line:

```
./phpstorm.sh
Info  | RdCoroutineScope          | 10:JobScheduler FJ pool 0/7 | RdCoroutineHost overridden 
2022-01-26 20:23:17,797 [   3154]   WARN - l.NotificationGroupManagerImpl - Notification group CodeWithMe is already registered (group=com.intellij.notification.NotificationGroup@758b06ac). Plugin descriptor: PluginDescriptor(name=Code With Me, id=com.jetbrains.codeWithMe, descriptorPath=plugin.xml, path=~/Applications/PhpStorm/plugins/cwm-plugin, version=213.5744.279, package=null, isBundled=true) 
WARN: Do not use URL connection as JarURLConnection
2022-01-26 20:23:18,402 [   3759]   WARN - .codestream.agent.AgentService - (node:82374) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead. 
2022-01-26 20:23:18,402 [   3759]   WARN - .codestream.agent.AgentService - (Use `codestream-agent.12.7.0+215 --trace-deprecation ...` to show where the warning was created) 
2022-01-26 20:23:18,767 [   4124]   WARN - tication.AuthenticationService - Auto sign-in failed: unable to retrieve token from password safe 

^C (this is me force-quitting the process from the terminal)

2022-01-26 20:23:32,122 [  17479]   WARN - pl.local.NativeFileWatcherImpl - Watcher terminated with exit code 130 
2022-01-26 20:23:32,139 [  17496]   WARN - .codestream.agent.AgentService - java.io.IOException: Stream closed 
org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:72)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.request(RemoteEndpoint.java:161)
	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:91)
	at com.sun.proxy.$Proxy73.shutdown(Unknown Source)
	at com.codestream.agent.AgentService.restart(AgentService.kt:180)
	at com.codestream.agent.AgentService$captureExitCode$1$1.invokeSuspend(AgentService.kt:292)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:740)
Caused by: java.io.IOException: Stream closed
	at java.base/java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:442)
	at java.base/java.io.OutputStream.write(OutputStream.java:157)
	at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
	at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:69)
	... 10 more
```

I can "fix" the problem by deleting the CodeStream plugin from `~/.local/share/JetBrains/PhpStorm2021.3/codestream-jb`. This is on Ubuntu 21.04, PHPStorm 2021.3.

This sounds like it might be similar to https://github.com/TeamCodeStream/codestream/issues/242, but that issue has been closed for months I figure this might be something new.

**Comment by tomgould** [comment]
This happened to me, I had to completely reinstall PHPStorm to get it to load again

**Comment by Kryptonien** [comment]
Very same problem here. I tried starting PhpStorm with `--trace-deprecation` as shown in the stacktrace. Here's my result.


```
phpstorm --trace-deprecation
Info  | RdCoroutineScope          | 14:JobScheduler FJ pool 3/15 | RdCoroutineHost overridden
2022-01-27 15:39:23,633 [   2308]   WARN - l.NotificationGroupManagerImpl - Notification group CodeWithMe is already registered (group=com.intellij.notification.NotificationGroup@7fb4705). Plugin descriptor: PluginDescriptor(name=Code With Me, id=com.jetbrains.codeWithMe, descriptorPath=plugin.xml, path=/opt/PhpStorm-213.6461.83/plugins/cwm-plugin, version=213.6461.83, package=null, isBundled=true)
WARN: Do not use URL connection as JarURLConnection
2022-01-27 15:39:24,049 [   2724]   WARN - .codestream.agent.AgentService - (node:24564) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
2022-01-27 15:39:24,051 [   2726]   WARN - .codestream.agent.AgentService - (Use `codestream-agent.12.7.0+215 --trace-deprecation ...` to show where the warning was created)
2022-01-27 15:39:24,098 [   2773]   WARN - agent.CodeStreamLanguageClient - [2022-01-27 14:39:24:087] Error(/home/ubuntu/.config/JetBrains/PhpStorm2021.3): /usr/bin/git -c core.quotepath=false -c color.ui=false rev-parse --show-toplevel)
    (128) Command failed: /usr/bin/git -c core.quotepath=false -c color.ui=false rev-parse --show-toplevel
fatal: not a git repository (or any of the parent directories): .git

fatal: not a git repository (or any of the parent directories): .git

2022-01-27 15:39:24,098 [   2773]   WARN - agent.CodeStreamLanguageClient - [2022-01-27 14:39:24:088] git — -c, core.quotepath=false, -c, color.ui=false, rev-parse, --show-toplevel,   cwd='/home/ubuntu/.config/JetBrains/PhpStorm2021.3'


Error: Command failed: /usr/bin/git -c core.quotepath=false -c color.ui=false rev-parse --show-toplevel
fatal: not a git repository (or any of the parent directories): .git

Error: Command failed: /usr/bin/git -c core.quotepath=false -c color.ui=false rev-parse --show-toplevel
fatal: not a git repository (or any of the parent directories): .git

    at ChildProcess.exithandler (node:child_process:397:12)
    at ChildProcess.emit (node:events:390:28)
    at ChildProcess.emit (node:domain:475:12)
    at maybeClose (node:internal/child_process:1064:16)
    at Socket.<anonymous> (node:internal/child_process:450:11)
    at Socket.emit (node:events:390:28)
    at Socket.emit (node:domain:475:12)
    at Pipe.<anonymous> (node:net:687:12)
2022-01-27 15:39:24,328 [   3003]   WARN - tication.AuthenticationService - Auto sign-in failed: unable to retrieve token from password safe
```

**Comment by planteater** [comment]
@quicksketch @tomgould @Kryptonien - Could I trouble you all to send a full log file? In your IDE go to Help > Collect Logs and Diagnostic Data. This will open the finder where you should see a newly created zip file. If you'd prefer not to share the files here, please send them to me at support@codestream.com. Thanks! We'll dig into this asap.

**Comment by Kryptonien** [comment]
I sent mine by email I hope it helps.

**Comment by marcelofarias** [comment]
@quicksketch you're running PHPStorm for Linux using WSL2 on Windows, right?

**Comment by quicksketch** [comment]
I'm running Ubuntu 21.10 and PHPStorm natively, not using WSL. I'll send
the logs via email.

**Comment by planteater** [comment]
Hi everyone - Just a quick update here. We're hoping to have a fix ready today. The issue is specific to the Linux / JxBrowser combination. Due to rendering problems JB has with JCEF on Linux (upvote this issue pls! https://youtrack.jetbrains.com/issue/IDEA-273222) we have to continue using JxBrowser for the time being.

In any case, in advance of our release, you should be able to workaround this for now by rolling back to v12.6 of the CodeStream extension.

**Comment by Kryptonien** [comment]
Thank you for the workaround.

However, do not forget to disable this option in the settings, otherwise it still does not work (at least for my case)
![image](https://user-images.githubusercontent.com/9591373/151824921-0c8a769e-4908-441f-954c-2b829f9de122.png)

**Comment by planteater** [comment]
Thanks for your patience everyone. This should be fixed in v12.8 of the CodeStream extension.
